apps/console/src/features/organizations/components/organization-attributes.tsx

```diff
@@ -0,0 +1,184 @@
+import { AlertLevels, SBACInterface, TestableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { useTrigger } from "@wso2is/forms";
+import { DynamicField, EmphasizedSegment, KeyValue, PrimaryButton } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useCallback, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Grid } from "semantic-ui-react";
+import { FeatureConfigInterface } from "../../core";
+import { patchOrganization } from "../api";
+import { OrganizationResponseInterface, PatchData } from "../models";
+
+interface OrganizationAttributesPropsInterface extends SBACInterface<FeatureConfigInterface>,
+    TestableComponentInterface {
+    /**
+     * Organizatin Data
+     */
+    organization: OrganizationResponseInterface;
+
+    /**
+     * Is Read Only access
+     */
+    isReadOnly: boolean;
+
+    /**
+     * On Attribute Update callback
+     */
+    onAttributeUpdate: (orgId: string) => void
+}
+
+export const OrganizationAttributes: FunctionComponent<OrganizationAttributesPropsInterface> = (
+    props: OrganizationAttributesPropsInterface
+): ReactElement => {
+
+    const {
+        organization,
+        isReadOnly,
+        onAttributeUpdate,
+        ["data-testid"]: testId
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch = useDispatch();
+
+    const [ submit, setSubmit ] = useTrigger();
+    const [ isSubmitting, setIsSubmitting ] = useState(false);
+
+    const updateOrgAttributes = useCallback((data: KeyValue[]) => {
+        setIsSubmitting(true);
+        const attributes = organization.attributes;
+
+        const updatedAttributes: PatchData[] = attributes
+            .map((attribute) => data.find((updated) => updated.key === attribute.key))
+            .filter((updated) => updated !== undefined)
+            .map((updated) => ({
+                operation: "REPLACE",
+                path: `/attributes/${updated.key}`,
+                value: updated.value
+            }));
```

Comment:
There are three loops here. I think we can reduce it to one.

Reply:
Agreed. Updated and Removed ones are merged into one, but kept addedAttributes as it is since we need to check the diff between updatedData with currentData